src/common/src/error.rs

```diff
@@ -131,6 +131,9 @@ pub enum ErrorCode {
 
     #[error("unrecognized configuration parameter \"{0}\"")]
     UnrecognizedConfigurationParameter(String),
+
+    #[error("StateTableError: {0}")]
+    StateTableError(String),
```

Comment:
Why do we include this variant? Shouldn't it be part of `StorageError` variant (https://github.com/risingwavelabs/risingwave/pull/5695/files#diff-750b1fee8849c1394669b6f99553c31e732840935ac37ecedc735dec0d7d6c9eR82)?

Reply:
🤔 Sounds reasonable